### PR TITLE
Moves securedrop-admin into Python 3

### LINKED ISSUE #3489: Migrate securedrop-admin to Python 3
## Description

We need to migrate away from Python 2 entirely, as it EOLs in 2020. This ticket is to migrate the `securedrop-admin` CLI tool to Python 3. The good news is that Python 3 is in Tails already (indeed the SecureDrop GUI updater in Tails uses Python 3), so this is a migration that can be done relatively soon.

### PR DESCRIPTION
**Pull request by kushaldas**
## Status

 Work in progress

## Description of Changes

Fixes #3489

This now needs an update of `prompt_toolkit` so that we can use it in Python3.5.

## Testing

In both Tails3, and Tails4, execute the following scenarios with prod vms/hardware:

- [ ] `./securedrop-admin sdconfig`
- [ ] `./securedrop-admin install`
- [ ] `./securedrop-admin tailsconfig`
- [ ] `./securedrop-admin backup`
- [ ] `./securedrop-admin restore`
- [ ] `./securedrop-admin update`
- [ ] `./securedrop-admin logs`
- [ ] `./securedrop-admin reset_admin_access`

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

### If you made changes to the server application code:

- [ ] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [x] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

### If you made changes to documentation:

- [x] Doc linting (`make docs-lint`) passed locally

### CODE REVIEW

**Comment by redshiftzero** [comment]
rebase issue?

**Comment by kushaldas** [comment]
There can be two different ways to migrate the virtualenv for the admins.

- After checking out `1.1.0`, if the admins run `./securedrop-admin install`, it will fail and ask to rerun `securedrop-admin setup` first. This is the (current implementation) and seems to be simpler.

- The other option is when `./securedrop-admin install` will see that there is a Python2 venv exists, and it will automatically do the setup work for Python3 venv. 

I am going to test the first item in a vm now.

**Comment by kushaldas** [comment]
`./securedrop-admin install` fails with following:

```

TASK [validate : Confirm host OS is Tails.] ************************************
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'ansible_lsb.major_release >= 9' failed. The error was: Unexpected templating type error occurred on ({% if ansible_lsb.major_release >= 9 %} True {% else %} False {% endif %}): unorderable types: AnsibleUnsafeText() >= int()"}

NO MORE HOSTS LEFT *************************************************************

NO MORE HOSTS LEFT *************************************************************
	to retry, use: --limit @/home/amnesia/Persistent/securedrop/install_files/ansible-base/securedrop-prod.retry

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1  
```

**Comment by conorsch** [comment]
@kushaldas I've appended a commit that resolves a few of the Ansible-related Python 3 issues. Tried most (but not all) of the test scenarios you outlined, and they've all been successful with the changes in place in Tails. For the record, I tested on Tails 3.16, with Python 3.5. Noticed some "weakref" stderr output in a few places, which is a significant cosmetic issue, but should not be a factor on Tails 4 / Python 3.7. 

Not approving, but I believe you should be unblocked for more vigorous testing now. Take another lap through it and see if you can identify any lingering Python 2 references that we haven't caught yet. When you're happy with it, I'd suggest we have @rmol and/or @zenmonkeykstop dig in for comprehensive review.

**Comment by kushaldas** [comment]
@rmol @zenmonkeykstop please go ahead with full scale review. This worked in my local testing.

**Comment by conorsch** [comment]
#4865 was just merged, introducing conflicts here. We choose to merge #4865 first to minimize the manual conflict resolution. @rmol / @zenmonkeykstop please give a rebase here, holler in gitter with any blockers!

**Comment by zenmonkeykstop** [comment]
On Tails 4.0~beta2 VM, checked out this branch and verified:

- [x] `./securedrop-admin setup`
- [x] `./securedrop-admin sdconfig`
- [x] `./securedrop-admin install` _(all good except for unmerged fixes in #4872)
- [x] `./securedrop-admin tailsconfig`
- [x] `./securedrop-admin backup`
- [x] `./securedrop-admin restore`
- [x] `./securedrop-admin update`
- [x] `./securedrop-admin logs`
- [x] `./securedrop-admin reset_admin_access`

**Comment by rmol** [comment]
Tested on Tails 3.16 and prod VMs:
    * [x]  `./securedrop-admin sdconfig`
    * [x]  `./securedrop-admin install`
    * [x]  `./securedrop-admin tailsconfig`
    * [x]  `./securedrop-admin backup`
    * [x]  `./securedrop-admin restore`
    * [x]  `./securedrop-admin update`
    * [x]  `./securedrop-admin logs`
    * [x]  `./securedrop-admin reset_admin_access`

**Comment by conorsch** [comment]
Rebased to resolve conflicts introduced via merge of #4865. The sole conflict was:

```
diff --cc admin/requirements.in
index 3e7fda7e0,2eebee385..000000000
--- a/admin/requirements.in
+++ b/admin/requirements.in
@@@ -1,2 -1,2 +1,7 @@@
++<<<<<<< HEAD
 +prompt_toolkit
 +pyyaml>=5.1.2
++=======
+ prompt_toolkit==2.0.9
+ pyyaml
++>>>>>>> 3a15f0e13... Moves securedrop-admin to Python3
```

which was manually solved as:

```
$ cat admin/requirements.in
prompt_toolkit==2.0.9
pyyaml>=5.1.2
```

**Comment by zenmonkeykstop** [comment]
On Tails 3.15 VM, checked out the rebased branch and verified:

- [x] `./securedrop-admin setup`
- [x] `./securedrop-admin sdconfig`
- [x] `./securedrop-admin install` 
- [x] `./securedrop-admin tailsconfig`
- [x] `./securedrop-admin backup`
- [x] `./securedrop-admin restore`
- [x] `./securedrop-admin update`
- [x] `./securedrop-admin logs`
- [x] `./securedrop-admin reset_admin_access`

LGTM!

**Comment by emkll** [comment]
While testing #4877 ran into an issue that may be worth flagging: we should ensure the 1.0.0 -> 1.1.0 docs have admins run `./securedrop-admin setup` as part of the upgrade process to 1.1.0, as the SecureDrop GUI updater will not run if the checked out branch is python3 . This will affect the version after SecureDrop 1.1.0, and the 1.0.0 -> 1.1.0 update will work (but the user should get a prompt to run `./securedrop-admin setup`

File: admin/requirements.txt [hunk 1]
```diff
@@ -110,10 +101,10 @@ paramiko==2.4.2 \
     --hash=sha256:3c16b2bfb4c0d810b24c40155dbfd113c0521e7e6ee593d704e84b4c658a1f3b \
     --hash=sha256:a8975a7df3560c9f1e2b43dc54ebd40fd00a7017392ca5445ce7df409f900fcb \
     # via ansible
-prompt-toolkit==1.0.15 \
-    --hash=sha256:1df952620eccb399c53ebb359cc7d9a8d3a9538cb34c5a1344bdbeb29fbcc381 \
-    --hash=sha256:3f473ae040ddaa52b52f97f6b4a493cfa9f5920c255a12dc56a7d34397a398a4 \
-    --hash=sha256:858588f1983ca497f1cf4ffde01d978a3ea02b01c8a26a8bbc5cd2e66d816917
+prompt_toolkit==2.0.9 \
```
Comment (redshiftzero) [inline admin/requirements.txt]: was diff review performed here?

Comment (redshiftzero) [reply admin/requirements.txt]: Diff review has been performed (timeboxed to ~4 hr): https://github.com/freedomofpress/securedrop-debian-packaging/wiki/prompt_toolkit-1.0.15--2.0.9

**Pull request merged**